src/main/java/uk/gov/hmcts/reform/iacasenotificationsapi/domain/personalisation/appellant/letter/AppellantInternalCaseNonStandardDirectionPersonalisation.java

```diff
@@ -0,0 +1,99 @@
+package uk.gov.hmcts.reform.iacasenotificationsapi.domain.personalisation.appellant.letter;
+
+import com.google.common.collect.ImmutableMap;
+import org.springframework.beans.factory.annotation.Value;
+import org.springframework.stereotype.Service;
+import uk.gov.hmcts.reform.iacasenotificationsapi.domain.entities.AsylumCase;
+import uk.gov.hmcts.reform.iacasenotificationsapi.domain.entities.AsylumCaseDefinition;
+import uk.gov.hmcts.reform.iacasenotificationsapi.domain.entities.Direction;
+import uk.gov.hmcts.reform.iacasenotificationsapi.domain.entities.DirectionTag;
+import uk.gov.hmcts.reform.iacasenotificationsapi.domain.entities.ccd.callback.Callback;
+import uk.gov.hmcts.reform.iacasenotificationsapi.domain.personalisation.LetterNotificationPersonalisation;
+import uk.gov.hmcts.reform.iacasenotificationsapi.domain.service.DirectionFinder;
+import uk.gov.hmcts.reform.iacasenotificationsapi.infrastructure.CustomerServicesProvider;
+
+import java.time.LocalDate;
+import java.time.format.DateTimeFormatter;
+import java.util.*;
+import java.util.stream.Collectors;
+
+import static java.util.Objects.requireNonNull;
+import static uk.gov.hmcts.reform.iacasenotificationsapi.domain.utils.AsylumCaseUtils.*;
+
+@Service
+public class AppellantInternalCaseNonStandardDirectionPersonalisation implements LetterNotificationPersonalisation {
+    private final String appellantInternalCaseNonStandardDirectionLetterTemplateId;
+    private final CustomerServicesProvider customerServicesProvider;
+    private final DirectionFinder directionFinder;
+
+    public AppellantInternalCaseNonStandardDirectionPersonalisation(
+            @Value("${govnotify.template.nonStandardDirectionInternal.appellant.letter}") String appellantInternalCaseNonStandardDirectionLetterTemplateId,
+            CustomerServicesProvider customerServicesProvider,
+            DirectionFinder directionFinder
+    ) {
+        this.appellantInternalCaseNonStandardDirectionLetterTemplateId = appellantInternalCaseNonStandardDirectionLetterTemplateId;
+        this.customerServicesProvider = customerServicesProvider;
+        this.directionFinder = directionFinder;
+    }
+
+    @Override
+    public String getTemplateId() {
+        return appellantInternalCaseNonStandardDirectionLetterTemplateId;
+    }
+
+    @Override
+    public Set<String> getRecipientsList(final AsylumCase asylumCase) {
+        return switch (isAppellantInUK(asylumCase)) {
```

Comment:
I think, here is a new utility function to do this.

`    public static Set<String> getAppellantAddressInCountryOrOoc(final AsylumCase asylumCase) {
`

Reply:
Done